usr/src/lib/libzfs/common/libzfs_sendrecv.c

```diff
@@ -609,11 +621,23 @@ send_iterate_snap(zfs_handle_t *zhp, void *arg)
 {
 	send_data_t *sd = arg;
 	uint64_t guid = zhp->zfs_dmustats.dds_guid;
+	uint64_t txg = zhp->zfs_dmustats.dds_creation_txg;
 	char *snapname;
 	nvlist_t *nv;
 
 	snapname = strrchr(zhp->zfs_name, '@')+1;
 
+	if (sd->parent_tosnap_txg != 0 && txg > sd->parent_tosnap_txg) {
```

Comment:
I agree with you. Please take a look at https://github.com/openzfs/openzfs/pull/76/commits/1a769a5a6c72158fd09b4e8d0fbb1f336f0639b7

Thank you!